ArcGISRuntimeSDKQt_CppSamples/Analysis/DistanceMeasurementAnalysis/DistanceMeasurementAnalysis.cpp

```diff
@@ -0,0 +1,188 @@
+  scene->setBaseSurface(surface);
+
+  // Add Analysis Overlay
+  AnalysisOverlay* analysisOverlay = new AnalysisOverlay(this);
+  m_sceneView->analysisOverlays()->append(analysisOverlay);
+
+  // Create and add the LocationDistanceMeasurement
+  const Point startLocation(-4.494677, 48.384472, 24.772694, SpatialReference::wgs84());
+  const Point endLocation(-4.495646, 48.384377, 58.501115, SpatialReference::wgs84());
+  m_distanceAnalysis = new LocationDistanceMeasurement(startLocation, endLocation, this);
+  m_distanceAnalysis->setUnitSystem(UnitSystem::Metric);
+  analysisOverlay->analyses()->append(m_distanceAnalysis);
+
+  // Set initial viewpoint
+  constexpr double distance = 200.0;
+  constexpr double pitch = 45.0;
+  constexpr double heading = 0.0;
+  constexpr double roll = 0.0;
+  const Camera initCamera(startLocation, distance, heading, pitch, roll);
+  const Viewpoint initViewpoint(startLocation, distance, initCamera);
+  scene->setInitialViewpoint(initViewpoint);
+
+  // Set the scene on the scene view
+  m_sceneView->setArcGISScene(scene);
+
+  connectSignals();
+}
+
+void DistanceMeasurementAnalysis::connectSignals()
+{
+  // connect to signal to obtain updated distances
+  connect(m_distanceAnalysis, &LocationDistanceMeasurement::measurementChanged, this, [this](Distance directDistance,
+                                                                                            Distance horizontalDistance,
+                                                                                            Distance verticalDistance)
+  {
+    const QString unitLabel = m_distanceAnalysis->unitSystem() == UnitSystem::Metric ? "m" : "ft";
+    m_directDistance = QString::number(directDistance.value(), 'f', 2) + QString(" %1").arg(unitLabel);
+    m_horizontalDistance = QString::number(horizontalDistance.value(), 'f', 2) + QString(" %1").arg(unitLabel);
+    m_verticalDistance = QString::number(verticalDistance.value(), 'f', 2) + QString(" %1").arg(unitLabel);
+    emit directDistanceChanged();
+    emit horizontalDistanceChanged();
+    emit verticalDistanceChanged();
+  });
+
+  // connect to mouse signals to update the analysis
+  connect(m_sceneView, &SceneQuickView::mousePressedAndHeld, this, [this](QMouseEvent mouseEvent)
+  {
+    m_isPressAndHold = true;
+    m_sceneView->screenToLocation(mouseEvent.x(), mouseEvent.y());
+  });
+
+  connect(m_sceneView, &SceneQuickView::mouseReleased, this, [this](QMouseEvent mouseEvent)
+  {
+    if (m_isNavigating)
+    {
+      m_isNavigating = false;
+      return;
+    }
+
+    if (mouseEvent.button() == Qt::RightButton)
+      return;
+
+    if (m_isPressAndHold)
+      m_isPressAndHold = false;
+    else
+      m_sceneView->screenToLocation(mouseEvent.x(), mouseEvent.y());
+  });
+
+  connect(m_sceneView, &SceneQuickView::mouseMoved, this, [this](QMouseEvent mouseEvent)
+  {
+    if (m_isPressAndHold)
+      m_sceneView->screenToLocation(mouseEvent.x(), mouseEvent.y());
```

Comment:
it seems like we might end up with multiple of these tasks in flight at once. Possibly you need to store the taskWatcher as a member and cancel any outstanding tasks or ignore the new events if one is already under way?